### ISSUE TITLE
[2.0] Error after update to latest code

### ISSUE THREAD

**Issue by 200Puls** [PolymerElements/iron-list#402]
### Description
After updating the code (Polymer and iron-list) to the latest version I see an error when the iron-lists items collection is rendered.

### Expected outcome

I would expect to see the list as before.

### Actual outcome
Uncaught TypeError: Cannot set property '__key__' of undefined
    at HTMLElement.<anonymous> (iron-list.html:1197)
    at HTMLElement._iterateItems (iron-list.html:1162)
    at HTMLElement._assignModels (iron-list.html:1192)
    at HTMLElement._update (iron-list.html:910)
    at HTMLElement._render (iron-list.html:1009)
    at HTMLElement.<anonymous> (iron-list.html:1432)
    at _timer._asyncModule.run (debounce.html:40)
    at web-animations-next-lite.min.js:16

### Live Demo

https://jsbin.com/sazojeqoci/2/edit?html,output

### Steps to reproduce

1. Put a `iron-list` element in the page.
2. Have a style for the list
3. Update the items array

### Browsers Affected

- [ x] Chrome

**Comment by blasten** [comment]
@keanulee would you like to give it a 👀?

**Comment by keanulee** [comment]
@blasten Seems like `_templateInstance` is renamed to `__templatizeInstance` in the latest Polymer

**Comment by frankiefu** [comment]
Is it related to this change: https://github.com/Polymer/polymer/pull/4339 ?

**Comment by blasten** [comment]
Yep. It is although that might introduce a problem for hybrid mode. A fix could be something like this:
```js
_getTemplateInstance(el) {
  return IS_V2 ? el.__templatizeInstance : el._templateInstance;
}
```

or maybe core can rename `_templateInstance` in V1. cc @kevinpschaaf

**Comment by kevinpschaaf** [comment]
The intention in core was to _privatize_ `_templateInstance` since it is an implementation detail of Templatize; sorry didn't realize `iron-list` was using it.  There's already a hybrid `modelForElement` API that provides the same information -- would using that be too onerous or perf-negative to use?

**Comment by blasten** [comment]
`modelForElement` should be fine. Thanks!

**Comment by keanulee** [comment]
This should be fixed by #403
